src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -721,11 +721,74 @@ private void modifyGenerator(Generator generator, GeneratorModificationInfos mod
         applyElementaryModifications(generator::setMaxP, generator::getMaxP, modificationInfos.getMaxActivePower(), subReporter, "Max active power");
         applyElementaryModifications(generator::setRatedS, generator::getRatedS, modificationInfos.getRatedNominalPower(), subReporter, "Rated nominal power");
         applyElementaryModifications(generator::setTargetP, generator::getTargetP, modificationInfos.getActivePowerSetpoint(), subReporter, "Active power set point");
-        applyElementaryModifications(generator::setTargetQ, generator::getTargetQ, modificationInfos.getReactivePowerSetpoint(), subReporter, "Reactive power set point");
         applyElementaryModifications(generator::setTargetV, generator::getTargetV, modificationInfos.getVoltageSetpoint(), subReporter, "Voltage set point");
-        applyElementaryModifications(generator::setVoltageRegulatorOn, generator::isVoltageRegulatorOn, modificationInfos.getVoltageRegulationOn(), subReporter, "Voltage regulation on");
 
-        // TODO connectivity modification
+        boolean isVoltageRegulationOn = modificationInfos.getVoltageRegulationOn() != null
```

Comment:
This whole function is really big and needs some comments, at least to facilitate its reading and/or to articulate the different parts of the treated form.